cnf-certification-test/lifecycle/suite.go

```diff
@@ -484,3 +490,51 @@ func testPodPersistentVolumeReclaimPolicy(env *provider.TestEnvironment) {
 		ginkgo.Fail(errMsg)
 	}
 }
+
+func testCPUIsolation(env *provider.TestEnvironment) {
+	ginkgo.By("Testing pods for CPU isolation requirements")
+
+	// Individual requirements we are looking for:
+	// - Resource Requests and Limits must be provided and identical. This determines whether or not we run the rest of the tests.
+	// - CPU Requests and Limits must be in the form of whole units
+	// - 'runtimeClassName' must be specified
+	// - Annotations must be provided disabling CPU and IRQ load-balancing
+
+	podsMissingIsolationRequirements := make(map[string]bool)
+
+	for _, put := range env.Pods {
+		it := isolation.NewCPUIsolationTester(put)
+
+		// Pods that have identical resources, are considered CPU isolated.
+		if it.LoadBalancingDisabled() {
+			if !it.AreCPUResourcesWholeUnits() {
+				errMsg := fmt.Sprintf("%s has been found to not have CPU resources that are whole units.", put.String())
+				logrus.Debugf(errMsg)
+				tnf.ClaimFilePrintf(errMsg)
+				podsMissingIsolationRequirements[put.Data.Name] = true
+			}
+
+			if !it.IsRuntimeClassNameSpecified() {
+				errMsg := fmt.Sprintf("%s has been found to not have runtimeClassName specified.", put.String())
+				logrus.Debugf(errMsg)
+				tnf.ClaimFilePrintf(errMsg)
+				podsMissingIsolationRequirements[put.Data.Name] = true
+			}
+
+			if !it.AreResourcesIdentical() {
+				errMsg := fmt.Sprintf("%s has been found to not have identical resources specified.", put.String())
+				logrus.Debugf(errMsg)
+				tnf.ClaimFilePrintf(errMsg)
+				podsMissingIsolationRequirements[put.Data.Name] = true
+			}
+		} else {
+			logrus.Debugf("%s is not an isolated pod. Skipping.", put.String())
```

Comment:
I would keep track of skipped pods and compliant pods using isolated cpus as well. At the end:
- if all the pods are skipped, no pods are compliant isolated, no pods are non compliant isolated -> skip the test because it does not apply for this CNF
- if some pods are skipped some are compliant/isolated and none is non-compliant -> pass the test
- if some pods are non compliant -> fail the test